proj: New Example with Audio Recording

svire-at opened the pull request:
* **Please check if the Pull Request fulfils these requirements**
  - [ ] PR title follows the style: `Type: Descriptive Name`
  - [ ] Commit messages follow the [guidelines](https://sensedu-shield.com/contributing/#styling-guidelines)
  - [ ] Code compiles
  - [ ] New additions are tested
  - [ ] Changes are documented on the [website](https://sensedu-shield.com) (refer to the [docs section](https://sensedu-shield.com/contributing/docs/))

* **Link related open issue**
#27

* **Provide a short description**


* **Provide a more detailed list of changes**
  - Item 1
  - Item 2
  - Item 3

* **Does this PR introduce a breaking change?** (What must be changed in other projects/examples due to this PR?)
N/A

* **Other information** (Anything else reviewers should know)
N/A

svire-at committed:
feat: Add Record_Audio example

Records audio and saves it in .wav file.

fanfare-ff committed:
refactor: Change the trigger for Record_Audio

Change buffer size to the multiple of 32-bytes
Move matlab trigger to request a certain amount of loops
Reduce hearable audio "cuts"

fanfare-ff reviewed (approved):
Looks good

I tried to tweak couple of things to reduce the audio "cuts", I pushed it to this branch
They are still hearable + noise, but maybe a little bit better

Before merging, review once more things I added and replace the recorded_audio.wav with the new one after these changes

svire-at committed:
feat: Checked and polished MATLAB script

Added audio recording visualization
Added new recorded audio example
Added example documentation
